### PR TITLE
fix(konflux): install operator-bundle dependencies as non-binaries

### PR DESCRIPTION
**Pull request by tommartensen**
### Description

change me!

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [ ] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

change me!

### CODE REVIEW

File: operator/bundle_helpers/README.md [hunk 1]
```diff
@@ -1,27 +1,54 @@
...
 
 ```bash
-poetry export -o requirements.txt
+pip-compile pyproject.toml --generate-hashes > requirements.txt
 ```
+
+2. Download pip_find_builddeps.py:
+
+```bash
+curl -O https://raw.githubusercontent.com/containerbuildsystem/cachito/master/bin/pip_find_builddeps.py
```
Comment (msugakov) [inline operator/bundle_helpers/README.md:39]:
```suggestion
curl -fO https://raw.githubusercontent.com/containerbuildsystem/cachito/master/bin/pip_find_builddeps.py
```

File: operator/bundle_helpers/README.md [hunk 2]
```diff
@@ -1,27 +1,54 @@
...
-
 For hermetic builds with Konflux, we need to provide the full list of resolved dependencies in `requirements.txt`.
 These will be prefetched with Cachi2 and made available to the container image build.
 
-We use Poetry to manage Python dependencies and for generating the complete `requirements.txt`.
-This tool can generate the full dependency tree for Python dependencies.
+## Python Dependency Management
+
+You need Python>=3.9 installed.
+We use Poetry to manage Python dependencies.
```
Comment (msugakov) [inline operator/bundle_helpers/README.md:9]: We can ditch `poetry` altogether and edit `pyproject.toml` by hand. We don't need `poetry.lock` because it does not participate in the following commands and it can be deleted.

Furthermore, since `pip-compile` can take `requirments.txt` format on the input, we can replace `pyproject.toml` with `requirements-konflux.in` like this:
```
PyYAML==6.0.2
```
That's it. We don't need `pytest` in Konflux context because `pytest` command is called by the `Makefile` which we bypass.

To bypass builder image update (for now), we can introduce `requirements-upstream.txt` with
```
PyYAML==6.0
pytest==7.0.1
```

To recap:
1. Let's please get rid of `poetry` mentions in this readme.
2. Please delete `poetry.lock`.
3. (optionally) Please switch from `pyproject.toml` to `requirements.txt` format (if that works).
4. (optionally) Create a separate `requirements.txt` to unfail GHA builds.

File: operator/bundle_helpers/README.md [hunk 3]
```diff
@@ -1,27 +1,54 @@
...
+- to add a new dependency: `poetry add PyYAML==6.0`
+- to update existing dependencies: `poetry update`
 
 For as long as we build downstream images with CPaaS, you should not add or use dependencies that aren't available OOTB in CPaaS in `render_templates` context.
 
 Check `pyproject.toml` for which dependencies and versions are managed with Poetry.
 
-To regenerate the `requirements.txt` file, run:
+## Prepare the fully resolved requirements files for Cachi2
+
```
Comment (msugakov) [inline operator/bundle_helpers/README.md:23]:
```suggestion

This must be done after any dependencies change for successful builds in Konflux.

```

File: operator/bundle_helpers/README.md [hunk 4]
```diff
@@ -1,27 +1,54 @@
...
+### Prerequisite
+
+- Install [pip-compile](https://pip-tools.readthedocs.io/en/stable/).
```
Comment (msugakov) [inline operator/bundle_helpers/README.md:26]: I installed pip-compile via `pipx` which somewhere was suggested as a recommended tool for such Python-based CLIs.

My sequence was
````bash
$ python3 -m pip install --user pipx
$ python3 -m pipx ensurepath
$ pipx install pip-tools
````

We could offer it as a suggestion of how to install the tool.

Comment (tommartensen) [reply operator/bundle_helpers/README.md]: https://github.com/stackrox/stackrox/pull/13265/commits/f8ea739d5ff4b1e91962ce518b5fb8940276859f

File: operator/bundle_helpers/README.md (hunk 3 above)
Comment (tommartensen) [reply operator/bundle_helpers/README.md:9]: https://github.com/stackrox/stackrox/pull/13265/commits/d2610265464b8e8cc79e6adb406c791cad237519

**Comment by tommartensen** [comment]
@msugakov see the commit status on https://github.com/stackrox/stackrox/pull/13265/commits/d2610265464b8e8cc79e6adb406c791cad237519 for green builds.

https://github.com/stackrox/stackrox/pull/13265/commits/f8ea739d5ff4b1e91962ce518b5fb8940276859f is only a README change.

File: operator/bundle_helpers/README.md [hunk 5]
```diff
@@ -1,27 +1,48 @@
 # Bundle Helpers
 
-## Python Dependency Management
-
 For hermetic builds with Konflux, we need to provide the full list of resolved dependencies in `requirements.txt`.
-These will be prefetched with Cachi2 and made available to the container image build.
+The dependency source files will be prefetched with Cachi2 and made available to the container image build.
+Because GHA/upstream is running on a different Python version, it uses `requirements-upstream.txt` to mange its own dependencies.
+Follow the procedure below after any dependencies change for successful builds in Konflux and mirror those changes into the `requirements-upstream.txt`.
```
Comment (msugakov) [inline operator/bundle_helpers/README.md:6]: After looking at five `requirements*` files in the `bundle_helpers/` directory, I realized we should do better job explaining what is what, at least for four not-gitignored files. Do you mind adding a short list like?

* `requirements.in` -> blah
* `requirements.txt` -> blah
* `requirements-build.txt` -> blah
* ...

File: operator/bundle_helpers/requirements-upstream.txt [hunk 6]
```diff
@@ -0,0 +1,2 @@
+PyYAML==6.0
```
Comment (msugakov) [inline operator/bundle_helpers/requirements-upstream.txt:1]: I suggest renaming this file to `requirements-gha.txt`. The idea is to move away from "upstream"/"downstream" terminology and division as "downstream" is hopefully going away.

I also suggest adding a `TODO(ROX-26860)` comment meaning we want to delete this file and switch to just `requirements.txt`.

File: operator/bundle_helpers/requirements.in [hunk 7]
```diff
@@ -0,0 +1 @@
+pyyaml==6.0.2
```
Comment (msugakov) [inline operator/bundle_helpers/requirements.in:1]: Sorry, I'd take back what I suggested before.
If we are to switch to a single `requirements.txt` file after ROX-26860, it could be better put `pytest` here than tediously describe in ROX-26860 all what needs to be done.

I check that adding pytest does not bring any modifications to `requirements-build.txt` so it should be smooth?

**Review by msugakov — COMMENTED** [review]
Looks good. Only a couple cosmetic things.

File: operator/bundle_helpers/requirements.in (hunk 7 above)
Comment (tommartensen) [reply operator/bundle_helpers/requirements.in:1]: it is in fact smooth. 
I updated ROX-26860 the remove the gha file once the Python version is bumped.
